metrics/store.go

```diff
@@ -26,13 +28,43 @@ func NewStore() (s *Store) {
 func (s *Store) Add(m *Metric) error {
 	s.Lock()
 	defer s.Unlock()
+	dupeIndex := -1
 	if len(s.Metrics[m.Name]) > 0 {
 		t := s.Metrics[m.Name][0].Kind
 		if m.Kind != t {
 			return errors.Errorf("Metric %s has different kind %s to existing %s.", m.Name, m.Kind, t)
 		}
+
+		// To avoid duplicate metrics:
+		// - copy old LabelValues into new metric;
+		// - discard old metric.
+		for i, v := range s.Metrics[m.Name] {
+			//
+			if v.Program != m.Program {
+				continue
+			}
+			if v.Type != m.Type {
+				continue
+			}
+			if reflect.DeepEqual(v.Keys, m.Keys) {
```

Comment:
@jaqx0r I'm fixing this part up, but I'd appreciate your opinion on that. I spinned it on my test environment and it seems to do the game.

Reply:
This seems fine. :)  Thanks!